### PR TITLE
Adding autoscale docs for pg_num (BSC#1163693)

### PR DESCRIPTION
**Pull request by asettle**

### CODE REVIEW

File: xml/admin_ceph_datamgm.xml [hunk 1]
```diff
@@ -1451,30 +1451,88 @@ osdmap e13 pg 1.6c (1.6c) -> up [1,0] acting [1,0]
...
-  </sect2>
+     <title>Reverting Lost Objects</title>
+     <para>
+      If the cluster has lost one or more objects and you have decided to abandon
+      the search for the lost data, you need to mark the unfound objects as
+      'lost'.
+     </para>
+     <para>
+      If the objects are still lost after having queried all possible locations,
+      you may need to give up on the lost objects. This is possible given unusual
```
Comment (tbazant) [inline xml/admin_ceph_datamgm.xml]: This sentence is long and complicated, I don't understand it 100%

File: xml/admin_ceph_datamgm.xml [hunk 2]
```diff
@@ -1451,30 +1451,88 @@ osdmap e13 pg 1.6c (1.6c) -> up [1,0] acting [1,0]
...
+      were performed before the writes themselves are recovered.
+     </para>
+     <para>
+      Currently the only supported option is 'revert', which will either roll
+      back to a previous version of the object, or forget about it entirely in
+      case of a new object. To mark the 'unfound' objects as 'lost', run the
+      following:
+     </para>
+  <screen>
+  &prompt.root;ceph pg <replaceable>PG_ID</replaceable> mark_unfound_lost revert|delete
```
Comment (tbazant) [inline xml/admin_ceph_datamgm.xml]:
```suggestion
  &prompt.cephuser;ceph pg <replaceable>PG_ID</replaceable> mark_unfound_lost revert|delete
```

File: xml/admin_ceph_datamgm.xml [hunk 3]
```diff
@@ -1451,30 +1451,88 @@ osdmap e13 pg 1.6c (1.6c) -> up [1,0] acting [1,0]
...
+  </screen>
+    </sect2>
+    <sect2 xml:id="op-pgs-autoscaler">
+     <title>PG Auto-scaler</title>
+     <para>
+       As of the Nautilus release, &ceph; includes a new manager module called
+       <literal>pg_autoscaler</literal> that allows the cluster to consider the
+       amount of data actually stored (or expected to be stored) in each
+       pool and choose appropriate <literal>pg_num</literal> values automatically.
```
Comment (tbazant) [inline xml/admin_ceph_datamgm.xml]:
```suggestion
       pool and choose appropriate <option>pg_num</option> values automatically.
```

File: xml/admin_ceph_datamgm.xml [hunk 4]
```diff
@@ -1451,30 +1451,88 @@ osdmap e13 pg 1.6c (1.6c) -> up [1,0] acting [1,0]
...
+     <para>
+       The autoscaler is configured on a per-pool basis, and can run in
+       three modes:
+     </para>
+     <variablelist>
+       <varlistentry><term>warn</term>
+        <listitem>
+          <para>
+            In <emphasis>warn</emphasis> mode, a health warning is issued if
+            the suggested <literal>pg_num</literal> value is too different from the current value.
```
Comment (tbazant) [inline xml/admin_ceph_datamgm.xml]:
```suggestion
            the suggested <option>pg_num</option> value is too different from the current value.
```

File: xml/admin_ceph_datamgm.xml [hunk 5]
```diff
@@ -1451,30 +1451,88 @@ osdmap e13 pg 1.6c (1.6c) -> up [1,0] acting [1,0]
...
+            This is the default for new and existing pools.
+          </para>
+        </listitem>
+       </varlistentry>
+       <varlistentry><term>on</term>
+        <listitem>
+          <para>
+            In autoscaler <emphasis>on</emphasis> mode, the pool <literal>pg_num</literal> is
```
Comment (tbazant) [inline xml/admin_ceph_datamgm.xml]:
```suggestion
            In autoscaler <emphasis>on</emphasis> mode, the pool <option>pg_num</option> is
```

File: xml/admin_ceph_datamgm.xml [hunk 6]
```diff
@@ -1451,30 +1451,88 @@ osdmap e13 pg 1.6c (1.6c) -> up [1,0] acting [1,0]
...
+            adjusted automatically with no need for any administrator interaction.
+          </para>
+        </listitem>
+       </varlistentry>
+       <varlistentry><term>off</term>
+        <listitem>
+          <para>
+            The autoscaler can also be turned <emphasis>off</emphasis> for any given
+            pool, leaving the administrator to manage <literal>pg_num</literal> manually as before.
```
Comment (tbazant) [inline xml/admin_ceph_datamgm.xml]:
```suggestion
            pool, leaving the administrator to manage <option>pg_num</option> manually as before.
```

File: xml/admin_ceph_datamgm.xml [hunk 7]
```diff
@@ -1451,30 +1451,88 @@ osdmap e13 pg 1.6c (1.6c) -> up [1,0] acting [1,0]
...
+         want to exercise more manual control should use the <emphasis>warn</emphasis> mode.
+       </para>
+     </note>
+     <para>
+       To enable the autoscale for a particular pool:
+     </para>
+<screen>&prompt.cephuser;ceph osd pool set foo pg_autoscale_mode on</screen>
+     <para>
+       Once enabled, the current state of all pools and the proposed adjustments
+       can be queried via the CLI using the following command:
```
Comment (tbazant) [inline xml/admin_ceph_datamgm.xml]:
```suggestion
       can be queried on the command line:
```

File: xml/admin_ceph_datamgm.xml (hunk 1 above)
Comment (asettle) [reply xml/admin_ceph_datamgm.xml]: Hmm actually, I think lines 1462 - 1465 are kinda redundant. I think we could remove them. Thoughts?

Comment (tbazant) [reply xml/admin_ceph_datamgm.xml]: ok now

**Comment by asettle** [comment]
> This looks good to me and I'm happy to support things after some basic testing.
> 
> I'm wondering if we should at least mention the optional specification of expected pool size (see https://docs.ceph.com/docs/master/rados/operations/placement-groups/#specifying-expected-pool-size)? This can very well be solved in a different issue though.

That's entirely your call. I'd rather not remove this in a weeks time when we've decided to solve it with a different issue. Let me know :) otherwise, merging.

**Comment by jan--f** [comment]
> That's entirely your call. I'd rather not remove this in a weeks time when we've decided to solve it with a different issue. Let me know :) otherwise, merging.

Please go ahead. The details about expected sizes will certainly be an addition and not change any of this.

**Pull request merged**